src/controllers/vouch/delete-submission.ts

```diff
@@ -0,0 +1,62 @@
+import Joi from "@hapi/joi";
+import { recoverTypedSignature_v4 as recoverTypedSignatureV4 } from "eth-sig-util";
+import { RequestHandler } from "express";
+import requestMiddleware from "../../middleware/request-middleware";
+import { Vouch } from "../../models";
+
+const deleteVouchSchema = Joi.object().keys({
+  submissionId: Joi.string(),
+  signature: Joi.string(),
+});
+
+const deleteSubmission: RequestHandler = async (req, res) => {
+  const { submissionId, signature } = req.body;
+
+  if (!submissionId) {
+    res.status(401).json({
+      message: "No submissionId provided.",
+    });
+    return;
+  }
+  const normalizedSubmissionId = (submissionId as string).toLowerCase();
+
+  const msgParams = {
+    domain: {
+      name: "Proof Of Humanity",
+    },
+    message: {
+      claim: `Reapplying for proof of humanity. My address is ${normalizedSubmissionId}`,
+    },
+    primaryType: "Auth",
+    types: {
+      EIP712Domain: [{ name: "name", type: "string" }],
+      Auth: [{ name: "claim", type: "string" }],
+    },
+  };
+
+  const recoveredAddr = recoverTypedSignatureV4({
+    data: msgParams as any,
+    sig: signature as string,
+  });
+
+  const query: any = {};
+  if (submissionId) {
```

Comment:
Should we use `normalizedSubmissionId` here? As it's the variable used inside the if clause, and `submissionId` is already checked on line 15 and there are no updates to the variable.

Reply:
Not sure I follow it here, in short we need everything to be either checksummed or lower case. I picked lowercase because we've been using it lowercase in PoH due to the graph not handling checksummed addresses.

If we don't want to use `normalizedSubmissionId`, the api user could theoretically send the checksummed address and then authentication will fail (assuming the `recoveredAddr` is checksummed).

Reply:
Sorry! I meant if we should change `if (submissionId) {` to `if (normalizedSubmissionId) {` Since `if (submissionId)` is checked in line 15 and the function returns if it fails, and because inside the if we use `normalizedSubmissionId`.

Reply:
Oh I see now :+1: 

This is the leftover from copy-pasta. I've updated the code and got rid of the redundant if.